pkg/scheduler/cache/cache.go

```diff
@@ -1486,22 +1486,35 @@ func (sc *SchedulerCache) RecordJobStatusEvent(job *schedulingapi.JobInfo, updat
 	}
 	// Update podCondition for tasks Allocated and Pending before job discarded
 	for _, status := range []schedulingapi.TaskStatus{schedulingapi.Allocated, schedulingapi.Pending, schedulingapi.Pipelined} {
+		semaphore := make(chan struct{}, 16)
+		var wg sync.WaitGroup
 		for _, taskInfo := range job.TaskStatusIndex[status] {
 			// The pod of a scheduling gated task is given
 			// the ScheduleGated condition by the api-server. Do not change it.
 			if taskInfo.SchGated {
 				continue
 			}
 
-			reason, msg, nominatedNodeName := job.TaskSchedulingReason(taskInfo.UID)
-			if len(msg) == 0 {
-				msg = baseErrorMessage
-			}
-			if err := sc.taskUnschedulable(taskInfo, reason, msg, nominatedNodeName); err != nil {
-				klog.ErrorS(err, "Failed to update unschedulable task status", "task", klog.KRef(taskInfo.Namespace, taskInfo.Name),
-					"reason", reason, "message", msg)
-			}
+			wg.Add(1)
+			semaphore <- struct{}{}
```

Comment:
Maybe it is more better using goroutines pool to update those tasks concurent.

Reply:
Okay, I've already changed this using goroutines pool.